src/FSharp.Compiler.Interactive.Settings/fsiaux.fs

```diff
@@ -178,6 +181,11 @@ type InteractiveSession() =
 module Settings =
     let fsi = new InteractiveSession()
 
+    fsi.AddPrinter<FsiHelp.Parser.Help option>(fun help ->
+        match help with
+        | None -> "No help available"
```

Comment:
Does someone know why the `None` is not triggered? Something with the hosting  and reflection mechanism of the session?

Reply:
It seems to do that for any `'T option`:

```fsx
type T = T of int
fsi.AddPrinter (function Some (T t) -> $"Got a T: {t}." | None -> "Nope lol.")
let some = Some (T 1)
let none = Option<T>.None
```

```fsi
type T = | T of int
val some: T option = Got a T: 1.
val none: T option = None
```

It works for `'T voption`, though:

```fsx
type T = T of int
fsi.AddPrinter (function ValueSome (T t) -> $"Got a T: {t}." | ValueNone -> "Nope lol.")
let some = ValueSome (T 1)
let none = ValueOption<T>.ValueNone
```

```fsi
type T = | T of int
val some: T voption = Got a T: 1.
val none: T voption = Nope lol.
```

Reply:
Awesome, thanks a ton for figuring this out. We should get this documented.